pkg/controller/pxc/users.go

```diff
@@ -38,159 +36,350 @@ type ReconcileUsersResult struct {
 }
 
 func (r *ReconcilePerconaXtraDBCluster) reconcileUsers(cr *api.PerconaXtraDBCluster) (*ReconcileUsersResult, error) {
```

Comment:
What do you think about using `ctx` from `Reconcile` here and in all new functions instead of `context.TODO()`?

Reply:
I think that should be mandatory, but in this case, I just left everything like it was that I didn't need to change.